src/sparseml/pytorch/image_classification/utils/helpers.py

```diff
@@ -450,3 +282,129 @@ def save_model_training(
                 info_lines.append(f"{loss}: {val_res.result_mean(loss).item()}")
 
         info_file.write("\n".join(info_lines))
+
+
+# private methods
+def _get_collate_fn(arch_key: Optional[str] = None, task: Optional[Tasks] = None):
+    if not arch_key:
+        return None
+
+    is_ssd = "ssd" in arch_key.lower()
+    need_collate_function = (
+        is_ssd or "yolo" in arch_key.lower()
+    ) and task != Tasks.EXPORT
+    if need_collate_function:
+        return ssd_collate_fn if is_ssd else yolo_collate_fn
+    return None
+
+
+def _create_train_dataset_and_loader(
```

Comment:
can we combine _create_train and _create_val dataset and loader functions?

Reply:
Yes been meaning to do that and make the logic simpler, Will add in the upcoming helpers refactor